src/Crawler.php

```diff
@@ -447,7 +447,7 @@ public function startCrawling(UriInterface|string $baseUrl)
         }
     }
 
-    public function addToDepthTree(UriInterface $url, UriInterface $parentUrl, ?Node $node = null): ?Node
+    public function addToDepthTree(UriInterface $url, UriInterface $parentUrl, ?UriInterface $originalUrl = null, ?Node $node = null): ?Node
```

Comment:
Could you make this non-breaking by moving the new parameter to the last position?

Reply:
Sure!